src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -294,7 +294,7 @@ private async Task RunStepsAsync(List<IStep> embeddedSteps, ActionRunStage stage
                     // Evaluation error
                     Trace.Info("Caught exception from expression for embedded step.env");
                     step.ExecutionContext.Error(ex);
-                    step.ExecutionContext.Complete(TaskResult.Failed);
+                    SetStepConclusion(step, TaskResult.Failed);
```

Comment:
@thboop called this out as a bug, noting that `::Complete` shouldn't be called on a Composite step.